bin/main.rb

```diff
@@ -48,33 +57,49 @@ def turn
       puts 'Please enter a valid move'
       spot = gets.strip.to_i
     end
+    until @player.valid_move?(@board, input_to_index(spot))
+      puts "#{@current_player}, choose a spot between 1-9 which is not taken"
+      spot = gets.strip.to_i
+    end
+    move_board(input_to_index(spot))
+    puts `clear`
     display_board
+    # check if we have a winner or a draw
+    if @game.won?(@board)
+      puts "Congratulations #{@current_player} you won!!!" 
+      @game_over = true
+    elsif @game.full?(turn_count)
+      puts "It is a Draw!!!" 
+      @game_over = true
+    end
   end
 
   def turn_count
-    taken = 0
-    @board.each do |i|
-      taken += 1 unless i.nil?
-    end
-    taken
+    @board.select {|e| e == 'X' || e == 'O'}.size
   end
 
   def user_name
     2.times do |i|
       puts "Please enter name of player #{i + 1}:"
       @name << gets.strip
     end
+    until @name[1] != @name[0]
+      puts "Your name should be different than #{@name[0]}:"
+      @name[1] = gets.strip
+    end
```

Comment:
If the player class initializes the player object with  name and symbol, you could call that here